test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -83,12 +83,31 @@ func TestVaultDynamicSecret(t *testing.T) {
 		t.Parallel()
 	}
 
-	clusterName := os.Getenv("KIND_CLUSTER_NAME")
-	require.NotEmpty(t, clusterName, "KIND_CLUSTER_NAME is not set")
+	var clusterName string
+	if os.Getenv("SCALE_TESTS") != "" {
+		// When SCALE_TESTS is set, use EKS cluster
+		clusterName = os.Getenv("EKS_CLUSTER_NAME")
+		require.NotEmpty(t, clusterName, "EKS_CLUSTER_NAME is not set")
+	} else {
+		// Otherwise, use KIND cluster
+		clusterName = os.Getenv("KIND_CLUSTER_NAME")
+		require.NotEmpty(t, clusterName, "KIND_CLUSTER_NAME is not set")
+	}
 
 	operatorNS := os.Getenv("OPERATOR_NAMESPACE")
 	require.NotEmpty(t, operatorNS, "OPERATOR_NAMESPACE is not set")
 
+	// TODO: Extend this to support other dynamic create test cases
+	defaultCreate := 5 // Default count if no VDS_CREATE_COUNT is set
+	vdsCreateCount := getEnvInt("VDS_CREATE_COUNT", -1)
+	createOnlyCount := getEnvInt("VDS_CREATE_ONLY", defaultCreate)
+	mixedCount := getEnvInt("VDS_MIXED_CREATE", defaultCreate)
+
+	if vdsCreateCount != -1 {
```

Comment:
what if `vdsCreateCount` is less than -1 or 0?

Reply:
I updated the count so that we use the default count if it's <= 0.
We can either do this, or we can use the default count if < -1, and if 0 create no resources?